examples/numpyarrays/examply.nim

```diff
@@ -15,24 +15,47 @@ proc doStuff[T](el: T) : T {.inline.} =
 proc modArray*(x: NumpyArray[float64]) {.exportpy.} =
   # echo "modArrayInPlace.nim"
   # Example of accessing the buffer directly
-  x[0] = 123.0
-  x[1] = -5.0
+  x[0, 0] = 123.0
+  x[0, 1] = -5.0
 
 proc parallelForOp*(x: NumpyArray[float64]) : NumpyArray[float64] {.exportpy.} = 
   let np = pyImport("numpy")
   result = initNumpyArray[float64](x.shape)
+  let 
+    ur = result.toUnsafeView()
+    ux = x.toUnsafeView()
 
-  timeIt("parallelForLoop"):
-    for i in 0||(x.len-1):
-      result[i] = doStuff x[i]
+  for i in 0||(x.len-1):
+    ur[i] = doStuff ux[i] 
+
+proc parallelIndexedForOp*(x: NumpyArray[float64]) : NumpyArray[float64] {.exportpy.} = 
+  let np = pyImport("numpy")
+  result = initNumpyArray[float64](x.shape)
+  let 
+    ur = result.toUnsafeView()
+    ux = x.toUnsafeView()
+
+  for i in 0||(x.shape[0]-1):
+    for j in 0||(x.shape[1]-1):
+      result[i, j] = doStuff x[i, j]
```

Comment:
I don't think this compiles correctly, or does it? Last time I tried that, I ended up writing my own loop fusion macro https://github.com/Vindaar/llm.nim/blob/master/fuse_loops.nim.

Reply:
Just remembered that this was of course in the context of using an OpenMP `collapse` statement as well.

Reply:
And on the C backend in general `||` does not seem to compile at the moment due to `goto` statements. Only compiling with `--exceptions:quirky` makes it work. Maybe there's a different exception approach that also compiles, but well.